# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/EgeGln365/FlyRank_AI_Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## 1. My lane (or freestyle) and why

I chose **Lane 2: Refresh / Content Opportunity Scoring** because it focuses on prioritizing content pages for review using observable search, content, and engagement signals. This problem is both practical and challenging, as content teams often have limited time and resources to manually review every page. It also gives me the opportunity to strengthen my machine learning skills by comparing transparent baseline methods with models such as Logistic Regression, Decision Tree, Random Forest, and Gradient Boosting. My goal is to build an explainable decision-support system that helps editors prioritize pages rather than replacing human judgment.


In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## 2. The question: decision, action, cost of a wrong call

My research question is:

**Which content pages should be reviewed first based on observable search, content, and engagement signals?**

The goal is to improve the decision of prioritizing content pages for review when editors and SEO specialists have limited time and resources. Instead of manually inspecting every page, the system will generate a ranked queue of pages that are most likely to require attention.

The recommendations may include actions such as **Refresh, Expand, Protect, Prune, or Monitor**, but the final decision will always remain with a human reviewer.

A false positive may waste valuable editor time by recommending pages that do not require attention. A false negative may cause important declining or high-opportunity pages to be overlooked. Machine learning can support this decision because page performance depends on multiple interacting signals that are difficult to combine using simple rules alone.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
import os
import sys
import subprocess
import pandas as pd

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(
            ["git", "clone", "--depth", "1", REPO_URL, REPO_DIR],
            check=True
        )

    os.chdir(REPO_DIR)

print("Working directory:", os.getcwd())

csv_path = "data/raw/content_refresh_anonymized.csv"

assert os.path.exists(csv_path), (
    "Starter CSV not found. Make sure you are at the repository root."
)

df = pd.read_csv(csv_path)

print(f"Dataset size: {df.shape[0]:,} rows × {df.shape[1]} columns")
df.head(3)

Working directory: /content/flyrank-ml-internship-starter
Dataset size: 30,000 rows × 44 columns


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9


In [12]:
rows, cols = df.shape

print(f"Rows: {rows:,}")
print(f"Columns: {cols}")

Rows: 30,000
Columns: 44


In [11]:
df["trend_direction"].value_counts()

,count
trend_direction,
down,16262
stable,5962
up,4388
new,2236
flat,1152


In [10]:
df["trend_direction"].value_counts()

df["trend_direction"].value_counts(normalize=True) * 100

,proportion
trend_direction,
down,54.206667
stable,19.873333
up,14.626667
new,7.453333
flat,3.840000


The starter dataset contains 30,000 content pages from 32 pseudonymized clients. Among them, 16,262 pages (54.2%) are labeled as down, meaning that more than half of the pages show declining trends. These numbers suggest that manually reviewing every page would be inefficient, making a prioritization system for content review a valuable problem to study over the next several weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## 4. Careful words: what I can and can't claim

By the end of this project, my model will be able to rank content pages according to their review priority and recommend which pages should be reviewed first. It will provide decision-support recommendations based on historical search, content, and engagement signals, and its performance will be compared with a transparent baseline using evaluation metrics.

However, my work will not prove that refreshing a page will improve its performance, predict Google's ranking algorithm, or guarantee future traffic or ranking improvements. The recommendations are intended to support human decision-making rather than replace it.


In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.